### PR TITLE
Added MIT Scene Parse 150 dataset

### PR DESCRIPTION
**Pull request by nvnmo**
Added MIT Scene Parse 150 dataset as requested in #289 .

### CODE REVIEW

File: tensorflow_datasets/image/scene_parse_150.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+            num_shards=10,
+            gen_kwargs={
+              "images_dir_path" : os.path.join(dl_paths["images"], "validation"),
+              "annotations_dir_path": os.path.join(dl_paths["annotations"], "validation")
+            },
+        ),
+    ]
+
+  def _generate_examples(self, images_dir_path, annotations_dir_path):
+
```
Comment (us) [inline tensorflow_datasets/image/scene_parse_150.py]: delete unnecessary empty lines please

File: tensorflow_datasets/image/scene_parse_150.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+import tensorflow as tf
+import tensorflow_datasets as tfds
+
+# BibTeX citation
```
Comment (us) [inline tensorflow_datasets/image/scene_parse_150.py]: You can delete this line

File: tensorflow_datasets/image/scene_parse_150.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+_CITATION = """
+@inproceedings{zhou2017scene,
+    title={Scene Parsing through ADE20K Dataset},
+    author={Zhou, Bolei and Zhao, Hang and Puig, Xavier and Fidler, Sanja and Barriuso, Adela and Torralba, Antonio},
+    booktitle={Proceedings of the IEEE Conference on Computer Vision and Pattern Recognition},
+    year={2017}
+}
+@article{zhou2016semantic,
```
Comment (us) [inline tensorflow_datasets/image/scene_parse_150.py]: Why are you added 2016 citation. You use 2017 data.

**Comment by us** [comment]
Thank you for dataset, it's cool 😎

File: tensorflow_datasets/image/scene_parse_150.py (hunk 1 above)
Comment (nvnmo) [reply tensorflow_datasets/image/scene_parse_150.py]: Done!

File: tensorflow_datasets/image/scene_parse_150.py (hunk 2 above)
Comment (nvnmo) [reply tensorflow_datasets/image/scene_parse_150.py]: Done!

File: tensorflow_datasets/image/scene_parse_150.py (hunk 3 above)
Comment (nvnmo) [reply tensorflow_datasets/image/scene_parse_150.py]: Removed the old one.

File: tensorflow_datasets/image/__init__.py [hunk 4]
```diff
@@ -47,3 +47,4 @@
 from tensorflow_datasets.image.sun import Sun397
 from tensorflow_datasets.image.svhn import SvhnCropped
 from tensorflow_datasets.image.voc import Voc2007
+from tensorflow_datasets.image.scene_parse_150 import SceneParse150
```
Comment (us) [inline tensorflow_datasets/image/__init__.py]: Order should be alphabetical.

Comment (nvnmo) [reply tensorflow_datasets/image/__init__.py]: Sorted it

**Comment by nvnmo** [comment]
Ran pylint on all the files that changed. Are we following the line length rule from pep8?

**Comment by us** [comment]
You should follow the [tf code style guide](https://www.tensorflow.org/community/contribute/code_style).

**Comment by nvnmo** [comment]
I have formatted the code to tf code style. Kindly review it.

**Comment by us** [comment]
You have too much commits to single dataset. You can close this pr, squash commits and open a clean new pr.

**Comment by nvnmo** [comment]
@us opened a new PR #398 and squashed the commits.

**Pull request closed without merging**